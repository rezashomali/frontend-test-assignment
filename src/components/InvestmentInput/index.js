import React from "react";
import PropTypes from "prop-types";
import {
  FormControl,
  InputLabel,
  OutlinedInput,
  InputAdornment,
} from "@material-ui/core";

const InvestmentInput = ({ onInputChange, value }) => {
  return (
    <div>
      <FormControl fullWidth className="inputAmount" variant="outlined">
        <InputLabel htmlFor="standard-adornment-amount">Amount</InputLabel>
        <OutlinedInput
          type="number"
          error={value < 0 ? true : false}
          className="inputAmount__input"
          value={value}
          onChange={onInputChange}
          labelWidth={60}
          startAdornment={
            <InputAdornment className="inputAmount__symbol" position="start">
              â‚¬
            </InputAdornment>
          }
        />
        {value < 0 ? (
          <span className="inputAmount__error">incorect value</span>
        ) : (
          ""
        )}
      </FormControl>
    </div>
  );
};

InvestmentInput.propTypes = {
  value: PropTypes.number,
  onInputChange: PropTypes.func,
};

export default InvestmentInput;
