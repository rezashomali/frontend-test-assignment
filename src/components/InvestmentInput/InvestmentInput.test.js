import React from "react";
import renderer from "react-test-renderer";
import InvestmentInput from "./index";

test("InvestmentInput component snapshot ", () => {
  const component = renderer.create(
    <InvestmentInput value={1000} onInputChange={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
