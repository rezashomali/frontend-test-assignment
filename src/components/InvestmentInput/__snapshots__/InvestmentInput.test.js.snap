// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestmentInput component snapshot  1`] = `
<div>
  <div
    className="MuiFormControl-root inputAmount MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="standard-adornment-amount"
    >
      Amount
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root inputAmount__input MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root inputAmount__symbol MuiInputAdornment-positionStart"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          â‚¬
        </p>
      </div>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
        value={1000}
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-2"
          style={
            Object {
              "width": 53,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
    
  </div>
</div>
`;
